### PR TITLE
Post request

### PR DESCRIPTION
**Pull request by grace000**
This PR is a WIP for this [story](https://trello.com/c/5qL0ZIg7/4-post-request-375).

- Request router now sends requests to separate handlers based on the url.

### CODE REVIEW

File: src/main/java/github/grace000/erversay/Handlers/Handler.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package github.grace000.erversay.Handlers;
+
+import github.grace000.erversay.Request;
+
+public interface Handler {
```
Comment (BWoodfork) [inline src/main/java/github/grace000/erversay/Handlers/Handler.java]: This is a really generic name. What's it handling?

File: src/main/java/github/grace000/erversay/Handlers/Post.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class Post implements Handler {
+
+    public boolean isMethodAllowed(String method) {
+        return method.equals("POST");
+    }
+
+    public String handle(Request request) {
+        if (isMethodAllowed(request.method)) {
+            return new ResponseBuilder().withBody(request.body).withContentLength(request.body.length(), request.body).build();
+        } else return new ResponseBuilder().withStatus(NOT_FOUND_STATUS).build();
```
Comment (BWoodfork) [inline src/main/java/github/grace000/erversay/Handlers/Post.java]: This shouldn't all be on the same line.

File: src/main/java/github/grace000/erversay/Handlers/SimpleGet.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public String handle(Request request) {
+        if (isMethodAllowed(request.method)) {
+                return new ResponseBuilder().build();
+        } else return new ResponseBuilder().withStatus(NOT_FOUND_STATUS).build();
```
Comment (BWoodfork) [inline src/main/java/github/grace000/erversay/Handlers/SimpleGet.java]: Same here.

File: src/main/java/github/grace000/erversay/Handlers/OptionsTwo.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public String handle(Request request) {
+        if (isMethodAllowed(request.method)) {
+            return new ResponseBuilder().withHeaders(OPTIONS_TWO_HEADER).build();
+        } else return new ResponseBuilder().withStatus(NOT_FOUND_STATUS).build();
```
Comment (BWoodfork) [inline src/main/java/github/grace000/erversay/Handlers/OptionsTwo.java]: And here.

File: src/main/java/github/grace000/erversay/Handlers/Handler.java (hunk 1 above)
Comment (grace000) [reply src/main/java/github/grace000/erversay/Handlers/Handler.java]: I'm handling the different incoming paths. I'm assigning paths the handlers based on the name of the path. Then the handlers build different responses based on the incoming method that belongs to that path. I'll rename from handler to route handler.

**Pull request merged**